### PR TITLE
#1614 Cycloneplotter METplotlibDeprecationWarning observed in Matplotlib versions above 3.2

### LINKED ISSUE #1614: Cycloneplotter METplotlibDeprecationWarning observed in Matplotlib versions above 3.2
## Describe the Problem ##
During verification while migrating to Python 3.8.6, the MatplotlibDeprecationWarning has been noticed.  This warning is from within cartopy/mpl/geoaxes.py and corresponds to changes made in Matplotlib version 3.2 and above.

A simple fix to the code in the cycloneplotter wrapper can be accomplished by replacing this:
#Dummy point to add the additional label explaining the labelling of the first
        # point in the storm track
        plt.scatter(0, 0, zorder=2, marker=None, c='',
                    label="Date (dd/hhz) is the first " +
                            "time storm was able to be tracked in model")

with this
#Dummy point to add the additional label explaining the labelling of the first
        # point in the storm track
        colors = []
        plt.scatter(0, 0, zorder=2, marker=None, c=colors,
                    label="Date (dd/hhz) is the first " +
                            "time storm was able to be tracked in model")

A plot is still successfully created, but the log file contains this deprecation warning and eventually, when migrating to higher versions of Matplotlib, this will be an issue.

### Environment ###
Describe your runtime environment:
*1. Machine: (e.g. HPC name, Linux Workstation, Mac Laptop) Linux
*2. OS: (e.g. RedHat Linux, MacOS) Debian 
*3. Software version number(s) Python 3.8.6, cartopy 0.20.2

### To Reproduce ###
Describe the steps to reproduce the behavior:
*1. Run the cyclone plotter use case 
*2. Run is successful
*3. Open the log file 
*4. Scroll down and find the MatplotlibDeprecationWarning

/d1/personal/mwin/miniconda3/envs/metplus_py386/lib/python3.8/site-packages/cartopy/mpl/geoaxes.py:1546: MatplotlibDeprecationWarning: Using a string of single character colors as a color sequence is deprecated since 3.2 and will be removed two minor releases later. Use an explicit list instead.
  result = matplotlib.axes.Axes.scatter(self, *args, **kwargs)

### Relevant Deadlines ###
*List relevant project deadlines here or state NONE.*

### Funding Source ###
2792541

### Assignee ###
- [x] Select **engineer** Minna Win
- [x] Select **no scientist** required

### Labels ###
- [x] Select **component(s)**
- [x] Select **priority**
- [x] Select **requestor(s)**

### Projects and Milestone ###
- [x] Select **Organization** level **Project** for support of the current coordinated release
- [x] Select **Repository** level **Project** for development toward the next official release or add **alert: NEED PROJECT ASSIGNMENT** label
- [x] Select **Milestone** as the next bugfix version

## Define Related Issue(s) ##
Consider the impact to the other METplus components.
- [ ] [METplus](https://github.com/dtcenter/METplus/issues/new/choose), [MET](https://github.com/dtcenter/MET/issues/new/choose), [METdatadb](https://github.com/dtcenter/METdatadb/issues/new/choose), [METviewer](https://github.com/dtcenter/METviewer/issues/new/choose), [METexpress](https://github.com/dtcenter/METexpress/issues/new/choose), [METcalcpy](https://github.com/dtcenter/METcalcpy/issues/new/choose), [METplotpy](https://github.com/dtcenter/METplotpy/issues/new/choose)

## Bugfix Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [ ] Complete the issue definition above, including the **Time Estimate** and **Funding Source**.
- [ ] Fork this repository or create a branch of **main_\<Version>**.
Branch name: `bugfix_<Issue Number>_main_<Version>_<Description>`
- [ ] Fix the bug and test your changes.
- [ ] Add/update log messages for easier debugging.
- [ ] Add/update unit tests.
- [ ] Add/update documentation.
- [ ] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [ ] Push local changes to GitHub.
- [ ] Submit a pull request to merge into **main_\<Version>**.
Pull request: `bugfix <Issue Number> main_<Version> <Description>`
- [ ] Define the pull request metadata, as permissions allow.
Select: **Reviewer(s)** and **Linked issues**
Select: **Organization** level software support **Project** for the current coordinated release
Select: **Milestone** as the next bugfix version
- [ ] Iterate until the reviewer(s) accept and merge your changes.
- [ ] Delete your fork or branch.
- [ ] Complete the steps above to fix the bug on the **develop** branch.
Branch name:  `bugfix_<Issue Number>_develop_<Description>`
Pull request: `bugfix <Issue Number> develop <Description>`
Select: **Reviewer(s)** and **Linked issues**
Select: **Repository** level development cycle **Project** for the next official release
Select: **Milestone** as the next official version
- [ ] Close this issue.

### PR DESCRIPTION
**Pull request by bikegeek**
## Pull Request Testing ##

- [X] Describe testing already performed for these changes:</br>
- Followed the instructions for creating the cyclone track plots via the cycloneplotter wrapper:
https://metplus.readthedocs.io/en/develop/generated/model_applications/tc_and_extra_tc/Plotter_fcstGFS_obsGFS_ExtraTC.html#sphx-glr-generated-model-applications-tc-and-extra-tc-plotter-fcstgfs-obsgfs-extratc-py

and generated this plot:

- [x] Recommend testing for the reviewer(s) to perform, including the location of input datasets, and any additional instructions:</br>

on 'seneca', use the conda environment metplus_py386:

- _conda activate /home/met_test/.conda/envs/metplus_py386_
- clone the METplus repository and checkout the  _**feature_1614_matplotlib_deprecation**_ branch
- set up these environments, replace with your paths:
    - export MET_BUILD_BASE=/usr/local/met-10.0.0
    - export METPLUS_DATA=/d1/projects/METplus/METplus_Data
    - export PATH=${METPLUS_BUILD_BASE}/ush:${MET_BUILD_BASE}/bin:${PATH}
-  create a user_system.conf file (modify the OUTPUT_BASE path):
     

> [config]
>      OUTPUT_BASE = /d1/personal/mwin/feature_1614_matplotlib_deprecation/METplus/output
>       INPUT_BASE = {ENV[METPLUS_DATA]}
>       MET_INSTALL_DIR = /usr/local/met-10.0.0
> 

- run the use case:
    - python /d1/personal/mwin/feature_1614_matplotlib_deprecation/METplus/ush/run_metplus.py CyclonePlotter_fcstGFS_obsGFS_UserScript_ExtraTC.conf ./user_system.conf

- [x] Do these changes include sufficient documentation updates, ensuring that no errors or warnings exist in the build of the documentation? **[NA]**

- [x] Do these changes include sufficient testing updates? **[NA]**

- [x] Will this PR result in changes to the test suite? **[No]**</br>
If **yes**, describe the new output and/or changes to the existing output:</br>

- [x] Please complete this pull request review by **Before Beta2**.</br>

## Pull Request Checklist ##
See the [METplus Workflow](https://metplus.readthedocs.io/en/latest/Contributors_Guide/github_workflow.html) for details.
- [x] Add any new Python packages to the [METplus Components Python Requirements](https://metplus.readthedocs.io/en/develop/Users_Guide/overview.html#metplus-components-python-requirements) table.
- [x] Review the source issue metadata (required labels, projects, and milestone).
- [x] Complete the PR definition above.
- [x] Ensure the PR title matches the feature or bugfix branch name.
- [x] Define the PR metadata, as permissions allow.
Select: **Reviewer(s)**
Select: **Organization** level software support **Project** or **Repository** level development cycle **Project**
Select: **Milestone** as the version that will include these changes
- [ ] After submitting the PR, select **Linked issues** with the original issue number.
- [ ] After the PR is approved, merge your changes. If permissions do not allow this, request that the reviewer do the merge.
- [ ] Close the linked issue and delete your feature or bugfix branch from GitHub.

### CODE REVIEW

File: metplus/wrappers/cyclone_plotter_wrapper.py [hunk 1]
```diff
@@ -490,7 +490,8 @@ def create_plot(self):
 
         # Dummy point to add the additional label explaining the labelling of the first
         # point in the storm track
-        plt.scatter(0, 0, zorder=2, marker=None, c='',
+        colors = []
```
Comment (georgemccabe) [inline metplus/wrappers/cyclone_plotter_wrapper.py]: It looks like this variable is not used anywhere? Can it be removed?

**Comment by bikegeek** [comment]
I was going to remove it, but kept it in just in case we needed colors.
But I can remove it for now.
---------------
Minna Win
Pronouns: she/her
National Center for Atmospheric Research
Developmental Testbed Center
Phone: 303-497-8423
Fax:   303-497-8401
---------------

**Review by georgemccabe — APPROVED** [review]
I ran the use case on seneca successfully. It wasn't explicitly mentioned in the review instructions, but I assume that I am confirming that the deprecation warning is no longer present in the logs. This is the case, so I approve this PR given that there are no other actions I need to complete to confirm that the changes are sufficient.

**Comment by bikegeek** [comment]
Thanks, that is correct.

---------------
Minna Win
Pronouns: she/her
National Center for Atmospheric Research
Developmental Testbed Center
Phone: 303-497-8423
Fax:   303-497-8401
---------------

**Pull request merged**